paddle/fluid/operators/ngraph/ngraph_engine.cc

```diff
@@ -626,6 +631,21 @@ void NgraphEngine::Run(const framework::Scope& scope,
     }
   }
 
+  for (auto& op : fused_ops_) {
+    framework::RuntimeContext ctx(op->Inputs(), op->Outputs(), scope_);
+    if (op->Type() == "reshape2_grad") {
```

Comment:
why check reshape2 type in `Run` ? what if we have more ? transpose2 or others?

Reply:
reshape op adds a leading dimension of 0. https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/reshape_op.cc#L261

And chop it off in its grad op, https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/reshape_op.cc#L311.

In paddle the dimension 0 means placeholder or removable dimension, however ngraph will check dimension product and 0 will make ngraph treat the tensor as 0 element. So we have to process reshape2_grad op infershape differently. Doing this in ngraph engine can avoid modifying the original paddle file.